R/calculate_tdm.R

```diff
@@ -188,23 +190,37 @@ add_tdm <- function(data, groups) {
   data %>%
     group_by(!!!rlang::syms(groups)) %>%
     mutate(
-      .numerator = .data$scenario_production_plus_delta_t2 - .data$portfolio_production_plus_delta_t1,
+      .numerator_portfolio = .data$scenario_production_plus_delta_t2 - .data$portfolio_production_plus_delta_t1,
+      .numerator_scenario = .data$scenario_production_plus_delta_t2 - .data$scenario_production_plus_delta_t1,
       .denominator = .data$scenario_production_plus_delta_t2 - .data$scenario_production_t0,
-      # TODO: This case we
-      tdm_technology_value = ifelse(
+      tdm_technology_value_portfolio = ifelse(
         .data$.denominator == 0,
         0,
-        max(0, (.data$.numerator / .data$.denominator) * .data$monotonic_factor) * 2
+        max(0, (.data$.numerator_portfolio / .data$.denominator) * .data$monotonic_factor) * 2
       ),
-      .numerator = NULL,
+      tdm_technology_value_scenario = ifelse(
+        .data$.denominator == 0,
+        0,
+        max(0, (.data$.numerator_scenario / .data$.denominator) * .data$monotonic_factor) * 2
+      ),
+      .numerator_portfolio = NULL,
+      .numerator_scenario = NULL,
       .denominator = NULL
     ) %>%
-    select(.data$tdm_technology_value, all_of(groups)) %>%
+    select(.data$tdm_technology_value_portfolio, .data$tdm_technology_value_scenario, all_of(groups)) %>%
     distinct() %>%
+    pivot_longer(
+      cols = tidyr::starts_with("tdm_technology_value"),
```

Comment:
should this be (with a "_" at the end?
```r
cols = tidyr::starts_with("tdm_technology_value_"),
```

Reply:
never mind, I see you used `names_prefix` below

Reply:
actually that doesn't even matter either.... I just would have been slightly more specific in my `starts_with` selection, but no worries